app/javascript/components/subnet-form/index.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './subnet-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import { API } from '../../http_api';
+
+const SubnetForm = ({ recordId }) => {
+  const [{ initialValues, isLoading, fields }, setState] = useState({ isLoading: !!recordId, fields: [] });
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+
+  const loadSchema = (appendState = {}) => ({ data: { form_schema: { fields } } }) => {
+    if (!!recordId && appendState.initialValues.type === 'ManageIQ::Providers::Openstack::NetworkManager::CloudSubnet') {
+      Object.assign(fields[0], {isDisabled: true});
+      Object.assign(fields[1], {isDisabled: true});
+      Object.assign(fields[4], {isDisabled: true});
+    }
+    setState((state) => ({
+      ...state,
+      ...appendState,
+      fields,
+    }));
+  };
+
+  useEffect(() => {
+    if (recordId) {
+      API.get(`/api/cloud_subnets/${recordId}`).then((initialValues) => {
+        if(typeof initialValues.cloud_network_id ==="string") {
+          initialValues.cloud_network_id=Number(initialValues.cloud_network_id);
+        }
+        if(typeof initialValues.cloud_tenant_id ==="string") {
+          initialValues.cloud_tenant_id=Number(initialValues.cloud_tenant_id);
+        }
+        API.options(`/api/cloud_subnets?ems_id=${initialValues.ems_id}`).then(loadSchema({ initialValues, isLoading: false }));
+      });
+    }
+  }, [recordId]);
+
+  const onSubmit = (values) => {
+    API.get(`/api/providers/${values.ems_id}`).then(({ type }) => {
+      if (type === 'ManageIQ::Providers::Openstack::NetworkManager') {
+        if (values.ip_version === undefined) {
+          values.ip_version = '4';
+        }
+        if (values.dhcp_enabled === undefined) {
+          values.dhcp_enabled = false;
+        }
+        values.enable_dhcp = values.dhcp_enabled;
+        delete values.dhcp_enabled;
+        delete Object.assign(values, values.extra_attributes).extra_attributes;
```

Comment:
Both of these changes are to alter the values being submitted by the form so that the cloud subnet is created correctly (renaming `dchp_enabled` to `enabled_dhcp` and removing the `extra_attributes` object while moving its contents back directly into `values`).

The alternative to this is to change the form schema in the provider and the API to match what the back-end is expecting.